Add PostList Component Example for Dynamic Data Fetching

Cuscal-Ashman opened the pull request:
## `<PostList>`

The **PostList** component fetches data asynchronously from a public API and displays the results in a styled list. It includes a dual-layer rotating loader to provide clear visual feedback during data loading, and it gracefully handles error states. This component is an excellent example of real-world asynchronous data fetching using React hooks like `useState` and `useEffect`.

### Usage

Below is an example of how to use the **PostList** component in your MDX documentation:

```mdx
<PostList />

gkoberger commented:
Hey! Appreciate the contribution!! We'll check it out ASAP :)

Cuscal-Ashman commented on components/PostList/PostList.mdx:
```diff
@@ -0,0 +1,91 @@
+            }
+          `}
+        </style>
+        <div style={{ display: 'flex', flexDirection: 'column', alignItems: 'center', justifyContent: 'center', padding: '2.5rem' }}>
+          <div style={{ position: 'relative', width: '48px', height: '48px', marginBottom: '1rem' }}>
```
Resolved 👍

Cuscal-Ashman commented:
## What’s changed
- Added `PostList` React component with async fetch, loading, and error states.
- Replaced inline styles with Tailwind CSS classes.
- Added example usage in a parent component.
- Updated README with a clear, condensed overview and usage.

## How to test
- Render `<PostList />` in your app.
- Verify loading, error, and data display states.
- Confirm styles use Tailwind, no inline styles.
- Review updated README for clarity.

Cuscal-Ashman commented on components/PostList/PostList.mdx:
```diff
@@ -0,0 +1,91 @@
+  useEffect(() => {
+    const fetchPosts = async () => {
+      try {
+        const response = await fetch("https://jsonplaceholder.typicode.com/posts?_limit=5");
```
Great idea, @rafegoldberg! 

I’ve updated the `PostList` component to accept a `url` prop with a default value, so it can be customized easily when used. This makes it more flexible for different data sources.

rafegoldberg commented on components/PostList/readme.md:
```diff
@@ -0,0 +1,14 @@
+
+| Prop | Type   | Required | Default                                                     | Description                     |
+|------|--------|----------|-------------------------------------------------------------|---------------------------------|
+| url  | string | No       | `"https://jsonplaceholder.typicode.com/posts?_limit=5"`    | API endpoint to fetch posts from |
```
Minor stylistic suggestion:

```suggestion
| `url`  | string | No       | https://jsonplaceholder.typicode.com/posts?_limit=5    | API endpoint to fetch posts from |
```

rafegoldberg commented on .DS_Store:
This is a mac system file; can we delete this before we merge?

rafegoldberg reviewed (commented):
🕶️ Lookin’ good; thanks for the changes! Once we [delete the `.DS_Store` file](https://github.com/readmeio/marketplace/pull/15#discussion_r2255046889) I'll approve this should be good to merge.

rafegoldberg reviewed (approved):
Actually, [I just removed that `.DS_Store` file for ya](https://github.com/readmeio/marketplace/pull/15#discussion_r2255046889). You should be good to merge this at your convenience @Cuscal-Ashman!

rafegoldberg commented:
Ohhh hrmm, seems like [there's one outstanding lint issue](https://github.com/readmeio/marketplace/actions/runs/16758296638/job/47446957610?pr=15#step:4:33) that we should fix up first. (But my general approval stands.)

Cuscal-Ashman commented:
Hey @rafegoldberg, I fixed the remaining lint issues (single quotes + ellipsis), and just updated the PR. Thanks again for removing the .DS_Store file earlier! Let me know if anything else needs tweaking — otherwise, I think we’re good to merge. 🚀

trishaprile reviewed (approved):
@ashmetrics awesome, feel free to merge this in!

ashmetrics commented:
Hi @trishaprile / @rafegoldberg, I don’t have the option to merge this PR. Could you please merge it when you get a chance? Thanks!

rafegoldberg commented:
@ashmetrics—ahh, didn't know you didn't have those permissions. Merged for ya!
